metagpt/provider/openai_api.py

```diff
@@ -323,3 +325,29 @@ async def amoderation(self, content: Union[str, list[str]]):
     async def _amoderation(self, content: Union[str, list[str]]):
         rsp = await self.llm.Moderation.acreate(input=content)
         return rsp
+
+    @staticmethod
+    def __get_rpm(config) -> int:
+        limit_session_key = config.get("OPENAI_LIMIT_SESSION_KEY", "")
+        default_rpm = int(config.get("RPM", 10))
+        if len(limit_session_key) > 0:
+            try:
+                response = requests.get(
+                                        "https://api.openai.com/dashboard/rate_limits",
+                                        headers={'Authorization': f'Bearer {limit_session_key}'},
+                                        timeout=10,
+                                        proxies={'https': openai.proxy}
+                                        )
+            except Exception as e:
+                logger.error(f"Connection to api.openai.com failed:{e}.Setting rpm to default parameter.")
+                return default_rpm
+            
+            if response.status_code == 200:
+                limit_dict = json.loads(response.text)[config.openai_api_model]
+                return limit_dict['max_requests_per_1_minute']
```

Comment:
different model has different rpm, not just a value

Reply:
> different model has different rpm, not just a value
I have indexed response.text by config.openai_api_model

Reply:
> different model has different rpm, not just a value

I have indexed response.text by config.openai_api_model